### PR TITLE
Update plugin parent POM

### PR DESCRIPTION
**Pull request by basil**
CC @nhojpatrick

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,34 +4,29 @@
...
         <commons-lang3.version>3.12.0</commons-lang3.version>
         <jenkins.version>2.289.1</jenkins.version>
-        <java.level>8</java.level>
+        <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
     </properties>
     <name>commons-lang3 v3.x Jenkins Api Plugin</name>
     <description>Jenkins Api Plugin to provide org.apache.commons:commons-lang3:${commons-lang3.version}.
         Will break usage upon jars being provided by core jenkins.</description>
-    <url>https://github.com/jenkinsci/commons-lang3-api-plugin</url>
-    <dependencyManagement>
```
Comment (nhojpatrick) [inline pom.xml:22]: want to keep dependencyManagement section, as if anything uses it, it will enforce which version is used.

File: pom.xml [hunk 2]
```diff
@@ -4,34 +4,29 @@
...
     <artifactId>commons-lang3-api</artifactId>
-    <version>3.12.0.1-SNAPSHOT</version>
+    <version>${revision}${changelist}</version>
     <packaging>hpi</packaging>
     <properties>
+        <revision>3.12.0.1</revision>
+        <changelist>-SNAPSHOT</changelist>
         <commons-lang3.version>3.12.0</commons-lang3.version>
         <jenkins.version>2.289.1</jenkins.version>
-        <java.level>8</java.level>
```
Comment (nhojpatrick) [inline pom.xml:16]: want to ensure it's very clear it's java 8, so want to keep, so doesn't accidentally get upgraded to 11.

File: pom.xml [hunk 3]
```diff
@@ -49,9 +44,10 @@
         </license>
     </licenses>
     <scm>
-        <connection>scm:git:https://github.com/jenkinsci/${project.artifactId}-plugin.git</connection>
-        <developerConnection>scm:git:git@github.com:jenkinsci/${project.artifactId}-plugin.git</developerConnection>
-        <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
+        <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
+        <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
+        <url>https://github.com/${gitHubRepo}</url>
+        <tag>${scmTag}</tag>
```
Comment (nhojpatrick) [inline pom.xml:50]: from what i'm aware tag is added during a release so you don't want it in the development branch

File: pom.xml (hunk 2 above)
Comment (basil) [reply pom.xml:16]: I suggest you read [the plugin POM release notes](https://github.com/jenkinsci/plugin-pom/releases/tag/plugin-4.40).

File: pom.xml (hunk 3 above)
Comment (basil) [reply pom.xml:50]: Then propose a change to the [archetype](https://github.com/jenkinsci/archetypes), but do not block arbitrary plugin build update PRs.

File: pom.xml (hunk 1 above)
Comment (basil) [reply pom.xml:22]: Not a valid concern, but I lack the time to explain why.

**Comment by basil** [comment]
Maintainer did not respond after 4 days.

**Pull request closed without merging**